File: leetcode/all/python3/1100-1199/1189.py
from cmath import inf
from collections import Counter


class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        ct = Counter(text)
        ans = inf
        for c in ('b', 'a', 'l', 'o', 'n'):
            if c in ('b', 'a', 'n'):
                ans = min(ans, ct[c])
            else:
                ans = min(ans, ct[c] // 2)
        return ans

File: archived/leetcode/all/typescript/1189.ts
function maxNumberOfBalloons(text: string): number {
    const map_get = (map, k) => map[k] || 0
    // balloon
    let ans = Infinity
    const map = {}
    for (const c of text) {
        map[c] = map_get(map, c) + 1
    }
    ans = Math.min(
        map_get(map, 'b'),
        map_get(map, 'a'),
        map_get(map, 'n'),
        ~~(map_get(map, 'l') / 2),
        ~~(map_get(map, 'o') / 2)
    )
    return ans
}
